### ISSUE TITLE
Decide what UE4 Plugin to leverage

### ISSUE THREAD

**Issue by maishiroma** [maishiroma/Project_Brng_App#1]
From what I have been researching, it appears that there aren't too many options for leveraging external apps and UE4 aside from a select few options:
- https://github.com/HeliosOrg/SimpleDataIntegration
- https://github.com/monsieurgustav/UE4-OSC
- https://github.com/ufna/VaRest

For what I was reading, all of the ones but the first one are written in C++. However, the first one looks like is primarily a web application (or at least the example used a web app). Maybe that one is the most friendly in terms of how one can use that to integrate into a standalone app? They were able to run the external app on an ipad, but again, it was web based.

So I think the easiest way to get the ball rolling on this is the first option. Although in order to do a quick PoC we would need to build the project that was included in the docs first and I'm not sure if the plugin works on later versions of the engine (they used `4.11` for their demo).

Alternatively, we COULD move to that version (I have yet to do any dev yet) so this option isn't far fetched.

**Comment by maishiroma** [comment]
After doing some research, I think I have some pretty good information on this:

1. So for plugins, I think VaRest is the best choice because it is currently supported with the latest version of UE4 (4.25.4 is the one I have installed), as well as I found some nice tutorials that go over a high level of what to expect from using it. Plus, it is really easy to install, so I think that is the way to go.
- https://www.youtube.com/watch?v=AdEliltFRII
- https://www.youtube.com/watch?v=DifFzUQgVzo 

2. So it looks like we need an additional requirement for using this plugin; an API server. I did find some tutorials on making some from scratch, but I have no idea how long that is going to take + it sounds like a much bigger endevour. However, there are some projects I just found that lets you do a fake API server. Meaning, you can run a program that mocks an actual API server which takes in and receives data. There's a bunch on GitHub that do this, but these are the two that seem to be the most popular:
- https://github.com/typicode/json-server
-  https://github.com/typicode/jsonplaceholder

While I have yet to go into details on those fake API servers, if a longterm goal of this project is to make a self built API, I think that could be cool. However, to get a prototype working, I think it would be ok to look into these fake api servers so that I can get started working on the code logic.

In other words, I think we have something with what we got here! I'm going to be looking more and more into the fake API servers + more resources for API servers in the meantime, but I think we are onto something!

**Comment by maishiroma** [comment]
As for writing API servers, this is what I found:
- Most API servers seem to be written with node.js ,php and ruby
  - I did see one tutorial for golang? 

It seems to be pretty flexible though, so I guess it just depends on what one is most comfortable in?

Video Tutorials:
- https://www.youtube.com/watch?v=W5b64DXeP0o (golang)
- https://www.youtube.com/watch?v=fsCjFHuMXj0 (node.js)

**Comment by mathplusyou** [comment]
Thanks for all the info @maishiroma , I'll look into building out the API server.
